app/lib/grades/grades_service/grades_service.dart

```diff
@@ -494,7 +496,35 @@ enum PredefinedGradeTypes {
   oralParticipation,
   vocabularyTest,
   presentation,
-  other,
+  other;
+
+  String toUiString() {
+    return switch (this) {
+      PredefinedGradeTypes.schoolReportGrade => 'Zeugnisnote',
+      PredefinedGradeTypes.writtenExam => 'Schriftliche Prüfung',
+      PredefinedGradeTypes.oralParticipation => 'Mündliche Beteiligung',
+      PredefinedGradeTypes.vocabularyTest => 'Vokabeltest',
+      PredefinedGradeTypes.presentation => 'Präsentation',
+      PredefinedGradeTypes.other => 'Sonstiges',
+    };
+  }
+
+  Icon getIcon() {
+    return switch (this) {
+      PredefinedGradeTypes.schoolReportGrade =>
+        const Icon(Symbols.contract, fill: 1),
+      PredefinedGradeTypes.writtenExam =>
+        const Icon(Symbols.edit_document, fill: 1),
+      PredefinedGradeTypes.oralParticipation =>
+        const Icon(Symbols.record_voice_over, fill: 1),
+      PredefinedGradeTypes.vocabularyTest =>
+        const Icon(Symbols.text_rotation_none, fill: 1),
+      PredefinedGradeTypes.presentation =>
+        const Icon(Symbols.co_present, fill: 1),
+      PredefinedGradeTypes.other =>
+        const Icon(Symbols.other_admission, fill: 1),
+    };
+  }
```

Comment:
Not sure if we want to change the location where we make this switch statement. Later, we will also load dynamically the icon for custom grade types (created by the user).

Reply:
Leave it there for now, we'll figure something out in the future